Apps/GatewayApi/src/Services/UserProfileService.cs

```diff
@@ -503,17 +503,21 @@ private async Task<UserProfileModel> BuildUserProfileModelAsync(UserProfile user
             return userProfileModel;
         }
 
-        private DateTime? GetLatestTourChangeDateTime()
+        private async Task<DateTime?> GetLatestTourChangeDateTimeAsync(CancellationToken ct)
         {
-            return this.cacheProvider.GetOrSet<DateTime?>(
+            return await this.cacheProvider.GetOrSetAsync(
                 $"{TourApplicationSettings.Application}:{TourApplicationSettings.Component}:{TourApplicationSettings.LatestChangeDateTime}",
-                () =>
+                async () =>
                 {
-                    ApplicationSetting? applicationSetting = this.applicationSettingsDelegate.GetApplicationSetting(
+                    ApplicationSetting? applicationSetting = await this.applicationSettingsDelegate.GetApplicationSettingAsync(
                         TourApplicationSettings.Application,
                         TourApplicationSettings.Component,
-                        TourApplicationSettings.LatestChangeDateTime);
-                    return applicationSetting?.Value != null ? DateTime.Parse(applicationSetting.Value, CultureInfo.InvariantCulture).ToUniversalTime() : null;
+                        TourApplicationSettings.LatestChangeDateTime,
+                        ct);
+
+                    return applicationSetting?.Value != null
+                        ? DateTime.Parse(applicationSetting.Value, CultureInfo.InvariantCulture).ToUniversalTime()
+                        : (DateTime?)null;
```

Comment:
Q: I didn't think you would need to "Parse" null for a nullable type. Did this register as an error in IDE?

Reply:
Yes. There was an issue.

Reply:
Build Issue without the cast.

<img width="2020" alt="Screenshot 2023-12-22 at 4 36 58 PM" src="https://github.com/bcgov/healthgateway/assets/58790456/24c7283b-bb33-4af8-9a3b-6bab4efcfd12">

No Build issue with cast.

<img width="2045" alt="Screenshot 2023-12-22 at 4 38 17 PM" src="https://github.com/bcgov/healthgateway/assets/58790456/a111535c-4e0e-444a-b370-b5e0d2d2422b">